tests/integration/test_manifest_validation.py

```diff
@@ -0,0 +1,536 @@
+"""
+This module is responsible for running through the "Manifest Validation" portion of
+the schematic API test plan found here: <https://sagebionetworks.jira.com/wiki/spaces/SCHEM/pages/3055779846/Schematic+API+test+plan>.
+"""
+
+import json
+from typing import Dict
+
+import pytest
+import requests
+from flask.testing import FlaskClient
+from synapseclient.client import Synapse
+
+from tests.conftest import ConfigurationForTesting, Helpers
+
+EXAMPLE_SCHEMA_URL = "https://raw.githubusercontent.com/Sage-Bionetworks/schematic/develop/tests/data/example.model.jsonld"
+
+
+@pytest.fixture
+def request_headers(syn_token: str) -> Dict[str, str]:
+    """Simple bearer token header for requests"""
+    headers = {"Authorization": "Bearer " + syn_token}
+    return headers
+
+
+class TestManifestValidation:
+    @pytest.mark.local_or_remote_api
+    @pytest.mark.parametrize(
+        ("input_data_type", "input_file_name"),
+        [
+            ("Biospecimen", "mock_manifests/example_biospecimen_test.csv"),
+            (
+                "Patient",
+                "mock_manifests/TestManifestValidation_test_manifest_validation_basic_valid.csv",
+            ),
+        ],
+    )
+    def test_manifest_validation_basic_valid(
```

Comment:
@BryanFauble do we think we should add the same documentation that you've added in your test to document the tests here?

Reply:
I think that's a good idea. @jaymedina Is this something you'd be comfortable adding in?

An example that I did in another test:
```
        """
        Download a manifest from the Schematic API and verify that it is a valid Excel
        file. We are validating the following:
        - The first row of the Excel file contains the column headers
        - The first row is locked on scroll
        - Each cell in the first row has a comment/tooltip "TBD"
        - The "Family History" cell has a comment/tooltip with the expected message
        - The cell corresponding to "Sex" in Sheet1 has a dropdown list with values from Sheet2!B2:B4
        - The cell corresponding to "Diagnosis" in Sheet1 has a dropdown list with values from Sheet2!D2:D3
        - The cell corresponding to "Cancer Type" in Sheet1 has a dropdown list with values from Sheet2!F2:F6
        - The workbook contains two sheets: "Sheet1" and "Sheet2"
        - "Sheet2" is hidden
        - The values in "Sheet2" are as expected
        - The fill colors of the first row cells are as expected
        - Conditional formatting is functioning as expected
        Manual verification steps:
        - Open the Excel file prefixed with TestManifestGeneration_test_single_manifest_generation_excel.xlsx
        - When Diagnosis = "Cancer", [Cancer Type, Family History] is Light Blue (Required)
        """
```

Let's see if we can keep as much context about what each test is doing in the docstring.


@linglp had the great idea to be able to throw this information onto the pdoc site (https://sage-bionetworks.github.io/schematic/) so that we have:

1. A place to organize all of this information in 1 place
2. All changes are back by git, so we get history and tracking

Reply:
Of course! That's trivial, so no problem. I'll add it in the next commit. BTW are we planning to sunset the readthedocs documentation in place of pdocs?

Reply:
@andrewelamb do you know anything about the readthedocs page and the plans for it?

Reply:
I don't unfortunately. I'll ask Milen, he shoudl atleast know who to ask.